app/src/main/java/com/sunasterisk/movieintroduction/ui/base/BaseRecyclerAdapter.kt

```diff
@@ -0,0 +1,25 @@
+package com.sunasterisk.movieintroduction.ui.base
+
+import androidx.recyclerview.widget.RecyclerView
+
+abstract class BaseRecyclerAdapter<T, V : BaseViewHolder<T>> : RecyclerView.Adapter<V>() {
+
+    private val items = mutableListOf<T>()
+
+    override fun getItemCount() = items.size
+
+    override fun onBindViewHolder(holder: V, position: Int) {
+        getItem(position)?.let {
+            holder.onBindData(it)
+        }
+    }
+
+    protected fun getItem(position: Int): T? =
+        if (position in 0 until itemCount) items[position] else null
```

Comment:
Sao lại cần hàm getItem() nhỉ
```suggestion
    override fun onBindViewHolder(holder: V, position: Int) {
        holder.onBindData(items[position])
    }
```

Reply:
> Sao lại cần hàm getItem() nhỉ

Để khi lấy nó để lọc 1 list movie theo thể loại